lua/entities/magicbutton.lua

```diff
@@ -0,0 +1,463 @@
+    local lc = render.ComputeLighting(pos, self:GetUp())
+    local light1 = {}
+    light1.type = MATERIAL_LIGHT_POINT
+    light1.color = cvector * 15
+    light1.pos = pos
+    light1.fiftyPercentDistance = 2.35
+    light1.zeroPercentDistance = 4
+    render.SuppressEngineLighting(true)
+
+    for i = 0, 5 do
+        render.SetModelLighting(i, lc.x, lc.y, lc.z)
+    end
+
+    render.SetLocalModelLights({light1})
+
+    render.SetAmbientLight(0.05, 0.05, 0.05)
+    --render.SuppressEngineLighting( true )
+    self:DrawModel()
+    render.SuppressEngineLighting(false)
+end
+
+-- this is where we generate a random amount of points to give the player
+function MoneyPrize()
+    local basenumber = 100
+
+    for i = 1, math.random(10, 40) do
+        basenumber = (basenumber) ^ math.Rand(1.01, 1.06)
+    end
+
+    basenumber = math.Clamp(basenumber, 1000, math.random(100000, 500000))
+    basenumber = math.Round(basenumber, -3)
+
+    return basenumber * 1
+end
+
+local function MagicOutcomeBountyAndPrize(ply)
+    local amount = MoneyPrize()
+    if (ply.PS_GivePoints == nil or SetPlayerBounty == nil or GetPlayerBounty == nil) then return nil end
+    ply:PS_GivePoints(amount)
+    local add = GetPlayerBounty(ply) + amount
+    SetPlayerBounty(ply, add)
+
+    return amount
+end
+
+local function MagicOutcomeKleinerFanclub(ply)
+    KLEINER_OVERRIDE_TARGET = ply
+
+    timer.Create("KLEINER_GOD_EXPIRE", 60 * 15, 1, function()
+        KLEINER_OVERRIDE_TARGET = nil
+    end)
+
+    return ""
+end
+
+local function MagicOutcomeKleinerHatred(ply)
+    KLEINER_BULLIES[ply:SteamID()] = 5000
+
+    return ""
+end
+
+local function MagicOutcomeRandomTeleport(ply, button)
+    ply:SetPos(button:FindSuitableCastOrigin().StartPos)
+
+    return ""
+end
+
+local function MagicOutcomeKick(ply)
+    ply:Kick("Nice job! you found a secret button!")
+
+    return ""
+end
+
+local MagicButtonOutcomes = {
+    {
+        func = GiveMysteryBoxItem,
+        message = "and %s",
+        weight = 1
+    },
+    {
+        func = MagicOutcomeBountyAndPrize,
+        message = "and won %s points and also a %s point bounty on themself!",
+        weight = 1
+    },
+    {
+        func = MagicOutcomeKleinerFanclub,
+        message = "and won the attention of every kleiner on the map!",
+        weight = 1
+    },
+    {
+        func = MagicOutcomeKleinerHatred,
+        message = "and said an anti-kleiner slur! watch out!",
+        weight = 1
+    },
+    {
+        func = MagicOutcomeRandomTeleport,
+        message = "and teleported somewhere mysterious",
+        weight = 1
+    },
+    {
+        func = MagicOutcomeKick,
+        message = "and was kicked from the server",
+        weight = 1
+    },
+}
+
+--NOTE: On these functions, please return a string if there was a success, and nil if there was not.
+function ENT:Effect(ply)
+    local effect
+    local item
+    local counter = 0
+
+    while (item == nil and counter < 100) do
```

Comment:
Maybe you can shuffle the order of `MagicButtonOutcomes`, then iterate it, and return early if one of the effects "works"?